bson/src/test/unit-legacy/org/bson/BSONTest.java

```diff
@@ -55,57 +53,68 @@ public void tearDown() {
-                                 62, "cb7bad5697714ba0cbf51d113b6a0ee8");
-        checkEncodingAndDecoding(new BasicBSONObject("x", 4), 12, "d1ed8dbf79b78fa215e2ded74548d89d");
+                                 62,
+                "3e000000017800cdcccccccccc14400479002800000002300002000000610002310006000000656c696f740002"
+                        + "320002000000620008330001000a7a0000");
+        checkEncodingAndDecoding(new BasicBSONObject("x", 4), 12, "0c0000001078000400000000");
     }
 
     @Test
     public void testArray() throws IOException {
-        checkEncodingAndDecoding(new BasicBSONObject("x", new int[]{1, 2, 3, 4}), 41, "e63397fe37de1349c50e1e4377a45e2d");
+        checkEncodingAndDecoding(new BasicBSONObject("x", new int[]{1, 2, 3, 4}), 41,
+                "2900000004780021000000103000010000001031000200000010320003000000103300040000000000");
     }
 
     @Test
     public void testCode() throws IOException {
         BSONObject scope = new BasicBSONObject("x", 1);
         CodeWScope c = new CodeWScope("function() { x += 1; }", scope);
         BSONObject document = new BasicBSONObject("map", c);
-        checkEncodingAndDecoding(document, 53, "52918d2367533165bfc617df50335cbb");
+        checkEncodingAndDecoding(document, 53,
+                "350000000f6d6170002b0000001700000066756e6374696f6e2829207b2078202b3d20313b207d000c000000107800010000000000");
     }
 
     @Test
     public void testBinary() throws IOException {
-        byte[] data = new byte[10000];
-        for (int i = 0; i < 10000; i++) {
+        byte[] data = new byte[100];
+        for (int i = 0; i < 100; i++) {
             data[i] = 1;
         }
         BSONObject document = new BasicBSONObject("bin", data);
-        checkEncodingAndDecoding(document, 10015, "1d439ba5b959ecfe297a7862bf95bc10");
+        checkEncodingAndDecoding(document, 115,
+                "730000000562696e006400000000010101010101010101010101010101010101010101010101010101010101010101010101010101"
+                        + "01010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101"
+                        + "01010101010100");
     }
 
     private void checkEncodingAndDecoding(final BSONObject toEncodeAndDecode,
                                           final int expectedEncodedSize,
-                                          final String expectedHash) throws IOException {
+                                          final String expectedHex) throws IOException {
         // check encoding
         BSONEncoder bsonEncoder = new BasicBSONEncoder();
         OutputBuffer buf = new BasicOutputBuffer();
         bsonEncoder.set(buf);
         bsonEncoder.putObject(toEncodeAndDecode);
         assertEquals(expectedEncodedSize, buf.size());
-        assertEquals(expectedHash, hexMD5(buf.toByteArray()));
+        assertEquals(expectedHex, toHex(buf.toByteArray()));
```

Comment:
hexMD5 is no longer accessible.  The MD5 part of this was unnecessary though so I replaced it with just hex.

Reply:
Ack